Update isort script to match changes in the new release, isort v5.0.2

asheux opened the pull request:
Apparently, isort latest release ```isort v5.0.2``` is missing the ```--recursive or -rc``` flag and since the project uses the flag in it's script, it causes the tests to fail. An alternative to this is a dot, as in ```isort .```.

#### Missing flags
```
--apply
--recursive
```
#### Replacement
```
isort .
```

For more information on these changes see:
https://github.com/timothycrosley/isort/blob/develop/CHANGELOG.md#500-penny---july-4-2020

With this change, the build should be okay.

Commit:
Update isort script to match changes in the new release, isort v5.0.2

Commit:
Downgrade isort to version v4.3.21

Commit:
Add an alternative flag to --recursive in isort v5.0.2

asheux commented:
I'm trying to get rid of this error on travis ci build but I can seem to figure out how to make the build pass. Any ideas?

```py
ERROR: /home/travis/build/tiangolo/fastapi/fastapi/utils.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/fastapi/dependencies/utils.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/fastapi/openapi/models.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_additional_status_codes/test_tutorial001.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_security/test_tutorial006.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_security/test_tutorial001.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_security/test_tutorial003.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_security/test_tutorial005.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_wsgi/test_tutorial001.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_extra_models/test_tutorial003.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_extra_models/test_tutorial004.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_extra_models/test_tutorial005.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_additional_responses/test_tutorial001.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_additional_responses/test_tutorial003.py Imports are incorrectly sorted and/or formatted.

ERROR: /home/travis/build/tiangolo/fastapi/tests/test_tutorial/test_additional_responses/test_tutorial004.py Imports are incorrectly sorted and/or formatted.
```

asheux commented:
This was an easy fix. Apparently, with the new update in the command, the travis ci cache was updated so I had to run the formating script on my machine and push the changes to update travis cache.

Kludex commented on fastapi/__init__.py:
```diff
@@ -1,7 +1,6 @@
 """FastAPI framework, high performance, easy to learn, fast to code, ready for production"""
 
 __version__ = "0.58.1"
-
```
isort is removing this line? If yes, could you check why?

asheux replied:
I'm not sure but looking into it.

asheux replied:
This is being fixed by isort team. see https://github.com/timothycrosley/isort/issues/1283

Kludex commented on docs_src/nosql_databases/tutorial001.py:
```diff
@@ -1,11 +1,10 @@
 from typing import Optional
 
-from fastapi import FastAPI
```
We don't want to use isort in this tutorial, because `fastapi` and `pydantic` are really third party imports here.

asheux replied:
This is fixed

Kludex commented on tests/test_tutorial/test_additional_responses/test_tutorial001.py:
```diff
@@ -1,6 +1,5 @@
-from fastapi.testclient import TestClient
```
We don't want to use isort in this tutorial, because `fastapi` are really third party here.

asheux replied:
I'm going to git reflog to the commit before this so probably your review will be lost. I have new changes to fix all this

Kludex replied:
I think all of them use fastapi as third party... If that's true, yes.

asheux replied:
That's correct. I've made adjustment in the new isort configuration file. This should be flexible.

Kludex reviewed (commented):
Tutorial files should consider `fastapi` imports as third party.

Commit:
Add isort config file

tiangolo committed:
🚚 Import from docs_src for tests

tiangolo committed:
🎨 Format dependencies.utils

tiangolo committed:
🎨 Remove isort combine_as_imports, keep black profile

tiangolo committed:
🔧 Update isort config, use pyproject.toml, Black profile

tiangolo committed:
🔧 Update format scripts to use explicit directories to format

otherwise it would try to format venv env directories, I have several with different Python versions

tiangolo committed:
🎨 Format NoSQL tutorial after re-sorting imports

tiangolo committed:
🎨 Fix format for __init__.py

tiangolo commented:
Thanks for your contribution @asheux ! :cake: 

Thanks for the help here @Kludex :coffee: 

I just updated it to use the Black profile and moved the config to `pyproject.toml` to simplify it.

I also updated the tests for the tutorial examples so that they can detect the sources as first party, without having to disable isort for big sections of code.

Thanks for your contribution! :rocket:
